MLS.Agent/NativeAssemblyLoadHelper.cs

```diff
@@ -49,6 +54,119 @@ public void Handle(FileInfo assemblyFile)
             }
         }
 
+        private IEnumerable<string> ProbingFilenames(string name)
+        {
+            yield return name;                                                  // we can always try the name provided to pinvoke
+
+            if( !(Path.IsPathRooted(name)) )
+            {
+                var usePrefix = ProbingUsePrefix(name);
+                // Path is not rooted so we must try to add a prefix and suffix
+                foreach (var suffix in ProbingSuffixes())
+                {
+                    if (ProbingUseSuffix(name, suffix))
+                    {
+                        yield return ($"{name}{suffix}");
+                        if (usePrefix)
+                        {
+                            yield return ($"lib{name}{suffix}");
+                        }
+                    }
+                    else
+                    {
+                        yield return ($"lib{name}");
+                    }
+                }
+            }
+
+            // Probe for necessary suffixes --- these suffix' are platform specific
+            IEnumerable<string> ProbingSuffixes()
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    yield return ".dll";
+                    yield return ".exe";
+                }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    yield return ".dylib";
+                }
+                else
+                {
+                    yield return ".so";
+                }
+            }
+
+            // linux devs often append version # to libraries I.e mydll.so.5.3.2
+            bool ProbingUseSuffix(string name, string s)
+            {
+                return !(name.Contains(s + ".") || name.EndsWith(s));
+            }
+
+            // If the name looks like a path or a volume name then dont prefix 'lib'
+            bool ProbingUsePrefix(string name)
+            {
+                return name.IndexOf(Path.DirectorySeparatorChar) == -1              // If name has directory information no add no prefix
+                        && name.IndexOf(Path.AltDirectorySeparatorChar) == -1       // Alternate Directory seperator
+                        && name.IndexOf(Path.PathSeparator) == -1                   // Path seperator
+                        && name.IndexOf(Path.VolumeSeparatorChar) == -1;            // Volume seperator
+            }
+        }
+
+        private IEnumerable<string> ProbingPaths(string probingPath, string name)
```

Comment:
This function combines the probing root directory with the runtimes/rid/native to produce a list of dll's